### PR TITLE
Fixed wrong `ask` usage examples.

### PR DESCRIPTION
**Pull request by InFog**
As discussed in #1758

### CODE REVIEW

File: docs/documentation/getting-started/user-input/index.markdown [hunk 1]
```diff
@@ -7,7 +7,7 @@ User input can be required in a task or during configuration:
 
 ```ruby
 # used in a configuration
-set :database_name, ask('Enter the database name:')
+set :database_name, ask(:database_name, "default_database_name")
```
Comment (mattbrictson) [inline docs/documentation/getting-started/user-input/index.markdown]: The correct syntax would be:

```ruby
ask(:database_name, "default_database_name")
```

(no `set`)

File: docs/documentation/getting-started/user-input/index.markdown [hunk 2]
```diff
@@ -24,15 +24,16 @@ input from being displayed. This option should be used to ask the user for
 passwords and other sensitive data during a deploy run.
 
 ```ruby
-set :database_password, ask('Enter the database password:', 'default', echo: false)
+set :database_password, ask(:database_password, 'default_password', echo: false)
```
Comment (mattbrictson) [inline docs/documentation/getting-started/user-input/index.markdown]: Likewise here

**Comment by mattbrictson** [comment]
I would also mention something like this:

> Note that `ask` itself does not prompt the user immediately. The question is deferred until the first time `fetch` is used to obtain the setting. That means you can `ask` for many variables, but only the variables used by your task(s) will actually prompt the user for input.

The way the examples are written right now it is implied that `ask` immediately displays a prompt, which is not the case.

**Comment by InFog** [comment]
Ops, I forgot to remove the redundant `set` usages.

Done now. Also added the info about `ask` waiting for the first `fetch` to actually ask the question. I didn't know that to be honest :-) So, yes, it is good to point that out in the docs.

**Comment by mattbrictson** [comment]
Looks great, thanks! 🎉

**Comment by InFog** [comment]
:tada:

**Pull request merged**